src/wp-includes/post.php

```diff
@@ -7830,6 +7831,27 @@ function wp_queue_posts_for_term_meta_lazyload( $posts ) {
 			$post_type_taxonomies[ $post->post_type ] = get_object_taxonomies( $post->post_type );
 		}
 
+		foreach ( $post_type_taxonomies[ $post->post_type ] as $taxonomy ) {
+			$prime_post_terms[ $taxonomy ][] = $post->ID;
+		}
+	}
+
+	if ( $prime_post_terms ) {
+		$prime_term_ids = array();
+		foreach ( $prime_post_terms as $taxonomy => $post_ids ) {
+			$_term_ids = wp_cache_get_multiple( $post_ids, "{$taxonomy}_relationships" );
+			if ( is_array( $_term_ids ) ) {
+				$_term_ids      = array_filter( $_term_ids );
+				$prime_term_ids = array_merge( $prime_term_ids, ...$_term_ids );
+			}
+		}
+		if ( $prime_term_ids ) {
+			$prime_term_ids = array_unique( array_filter( $prime_term_ids ) );
```

Comment:
I first got a fatal error here with existing caches but after flushing the cache everything is working fine. Haven't further investigated this yet but I'm assuming it was just some corrupted cache because I'm not sure why there should be a WP_Term object instead of an ID?

```
Fatal error:  Uncaught Error: Object of class WP_Term could not be converted to string in /web/wp/wp-includes/post.php:7850
Stack trace:
#0 /web/wp/wp-includes/post.php(7850): array_unique(Array)
#1 /web/wp/wp-includes/class-wp-query.php(3494): wp_queue_posts_for_term_meta_lazyload(Array)
#2 /web/wp/wp-includes/class-wp-query.php(3749): WP_Query->get_posts()
#3 /web/wp/wp-includes/class-wp.php(663): WP_Query->query(Array)
#4 /web/wp/wp-includes/class-wp.php(783): WP->query_posts()
#5 /web/wp/wp-includes/functions.php(1332): WP->main('')
#6 /web/wp/wp-blog-header.php(16): wp()
#7 /web/index.php(17): require('/...')
#8 {main}
  thrown in /web/wp/wp-includes/post.php on line 7850
```

Reply:
Seems related to https://github.com/WordPress/wordpress-develop/commit/4a9f5fe3be47914b3f2b30aaf863761335cf1ad2 . Caches store an array of ids instead of terms. Have you upgraded flushed caches since 6.0.0 update?

Reply:
Thanks, that's indeed related. Turns out another plugin still stores the full objects, see https://github.com/polylang/polylang/pull/1154.

For maximum compatibility I have updated the code to do the same as in get_object_term_cache().